fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraLdp.java

```diff
@@ -763,6 +766,26 @@ private String mintNewPid(final String slug) {
         return pid;
     }
 
+    /**
+     * perform save with checking to prevent SN-Sibling nodes from creating
+     * @param resource the FedoraResource need to save
+     * @throws javax.jcr.AccessDeniedException
+     * @throws VersionException
+     * @throws LockException
+     * @throws ConstraintViolationException
+     * @throws RepositoryException
+     */
+    private synchronized void save(final FedoraResource resource) throws javax.jcr.AccessDeniedException,
```

Comment:
This is no better at all than what you had before. The keyword `synchronized` on an instance method just [locks on the monitor for the instance](https://docs.oracle.com/javase/tutorial/essential/concurrency/locksync.html). I do not know how much more clearly I can say this, but I will try once more: this class is a non-singleton JAX-RS resource. No two requests will be using the same instance. Synchronizing or locking on instance or static fields is _useless_ and _harmful_ here. Please remove this error.